analyses/upstream-analysis/util/function-run-QC.R

```diff
@@ -0,0 +1,66 @@
+###########################################################################
+#' Function to run miQC
+#' @param sce
+#'
+#' @return
+#' @export
+#'
+#' @examples
+run_miQC <- function(sce) {
+  if (any(sce$percent.mito == 0)) {
+    print("miQC did not work, use two steps filtering as defined by `params` for `min_genes` and `mtDNA_pct_default`, and the `FindOutlierThershold` function!")
+    } else {
+      print("miQC worked!")
+      model <- mixtureModel(sce)
+      print(is.nan(model@logLik))    # Apply is.nan function
+      parameters(model)
+      head(posterior(model))
+      print(plotModel(sce, model)) # Plot
+      print(plotFiltering(sce, model))
+      
+      # Convert sce back to seurat object 
+      filtered_sce <- as.Seurat(filterCells(sce, model), 
+                                counts = "counts",
+                                data = "logcounts")
+  }
+  return(filtered_sce)
+}
+
+
+###########################################################################
+#' Function to run QC with default parameters
+#' @param sce
+#'
+#' @return
+#' @export
+#'
+#' @examples
+run_QC_default <- function(sce) {
```

Comment:
It is a bit unclear why the function is called on sce as the parameter name, but it appears sce is not used within the function? Rather it seems like everything is done on the object called seurat_obj?

Reply:
We will use the `miQC` R package that works on `sce` objects for filtering. Therefore, the object was named accordingly. We convert back to a `seurat_obj` after that step in the QC script.

Reply:
Sorry, I should have been more clear in my comment. I see that the structure of the object used for the miQC function is an sce object. And the object structure for this function is a seurat object. I was referring more so to the naming and what the purpose of the sce parameter is. sce (the named input) is not used in this function, so it seems what the input the function is called with isn't relevant to what actual object the function is run with. It seems the function is being run with whatever object named seurat_obj exists outside the function, if that makes sense.